### ISSUE TITLE
Query about tps measurement

### ISSUE THREAD

**Issue by aflah02** [pytorch/torchtitan#2230]
Hi

I just wanted to confirm whether the tps logged during training considers the gradient accumulation steps? I couldn't figure it out from the code but maybe its because I'm new to the codebase. I would expect speed to go up as accumulation steps increase as you do a backward less however it doesn't seem to change much

**Comment by tianyu-l** [comment]
https://github.com/pytorch/torchtitan/blob/main/torchtitan/train.py#L613

Logging happens outside of and insensitive to grad accumulation.

> I would expect speed to go up as accumulation steps increase as you do a backward less however it doesn't seem to change much

With grad accumulation, you run N steps of forward + backward and 1 step of optimizer update. It won't save time, it saves memory.
